### PR TITLE
feat: zettaset padding

### PR DESCRIPTION
**Pull request by nicholasturner1**
Adding a command-line argument to specify a padding margin for all zettaset samples.

### CODE REVIEW

File: deepem/data/dataset/zettaset.py [hunk 1]
```diff
@@ -70,15 +78,30 @@ def convert_array(arr: ArrayLike) -> np.ndarray:
...
-            vol = sample.read_mask(key)[key]            
+            vol = sample.read_mask(key)[key]
             dset[name + "_mask"] = convert_array(vol).astype('uint8')
         else:
             dset[name + "_mask"] = np.ones_like(dset[name], dtype='uint8')
         print(f"{name + '_mask'}: {dset[name + '_mask'].shape}")
 
+        # applying padding
+        if zettaset_padding != (0, 0, 0):
+            widths = (
```
Comment (torms3) [inline deepem/data/dataset/zettaset.py:91]: This is always a source of confusion, but `zettaset_padding` is supposed to be in (x, y, z) order, right?
Both labels and their masks have been converted to (z, y, x) order in the above, so I guess the correct padding widths here should be:
```
widths = (
    (zettaset_padding[-1], zettaset_padding[-1]),
    (zettaset_padding[-2], zettaset_padding[-2]),
    (zettaset_padding[-3], zettaset_padding[-3]),
)
```

@nicholasturner1 Can you please confirm it?

**Review by torms3 — CHANGES_REQUESTED** [review]
Found some issues with (x, y, z) vs. (z, y, x) order when padding

File: deepem/data/dataset/zettaset.py (hunk 1 above)
Comment (nicholasturner1) [reply deepem/data/dataset/zettaset.py:91]: I made this argument use the same zyx convention as `fov`, `outputsz`, etc., so I think the current code should be correct.

File: deepem/data/dataset/zettaset.py [hunk 2]
```diff
@@ -46,12 +47,19 @@ def convert_array(arr: ArrayLike) -> np.ndarray:
 
     dset: dict[str, np.ndarray] = {}
 
-    # Image    
+    # Image
+    if zettaset_padding == (0, 0, 0):
+        image_bbox = sample.bbox
+    else:
+        image_bbox = Bbox(
+            sample.bbox.minpt - zettaset_padding, sample.bbox.maxpt + zettaset_padding
```
Comment (torms3) [inline deepem/data/dataset/zettaset.py]: If `zettaset_padding` uses the `zyx` convention, then I think you should convert it to the `xyz` order here?

Comment (nicholasturner1) [reply deepem/data/dataset/zettaset.py]: Ah, yeah good catch. I'll update that.

**Comment by nicholasturner1** [comment]
I've fixed the bbox modification error and made the image CloudVolume unbounded for cases when the padding requests data beyond the image boundaries. Implementing padding by reflection for this case seems unnecessary for now, but let me know if you disagree.

**Comment by torms3** [comment]
> I've fixed the bbox modification error and made the image CloudVolume unbounded for cases when the padding requests data beyond the image boundaries. Implementing padding by reflection for this case seems unnecessary for now, but let me know if you disagree.

Yeah I agree it's unnecessary.

**Pull request merged**